ValkeyJSON.md

```diff
@@ -0,0 +1,214 @@
+* Restricted syntax – Has limited query capabilities, compatible with RedisJSON v1.
+
+If a query path starts with '$', the enhanced syntax is used. Otherwise, the restricted syntax is used. A query using 
+the enhanced syntax always returns an array of values, while a restricted-syntax query always returns a single value. 
+
+![](./images/jsonpath_syntax.png)
+
+For details and examples, see https://docs.aws.amazon.com/memorydb/latest/devguide/json-document-overview.html#json-path-syntax.
+
+### ACL
+
+ValkeyJSON introduces a new ACL category - @json. The category has 22 JSON commands. No existing Valkey commands 
+are members of the @json category. 
+
+There are 4 existing ACL categories which are updated to include new JSON commands: @read, @write, @fast, @slow. The 
+table below has a column for each of these categories and a row for each command. If the cell contains a “y” then that 
+command must be added into that category. All of the other command members of those categories remain unchanged.
+
+![](./images/json_command_acl.png)
+
+### Info metrics
+
+Info metrics are visible through the “info json” or “info modules” command.
+
+| Name                    | 	Description                                                           |
+|:------------------------|:---------------------------------------------------------------------|
+| json_total_memory_bytes | 	Total amount of memory allocated to JSON documents and meta data. |
+| json_num_documents	  | Number of JSON keys                                                   | 
+
+### Configs
+
+![](./images/json_configs.png)
+
+### Document Size Limit
+
+JSON documents are stored internally in a format optimized for rapid access and modification. This format typically 
+results in consuming somewhat more memory than the equivalent serialized representation of the same document does. 
+The consumption of memory by a single JSON document is limited to 64MB by default, which is the size of the in-memory 
+data structure, not the serialized JSON string. The value can be overridden by module config json.max-document-size.
+
+The amount of memory consumed by a JSON document can be inspected by 
+using the “JSON.DEBUG MEMORY" or “MEMORY USAGE” command. “JSON.DEBUG MEMORY <key> [<path>]" can also report the size 
+of a JSON sub-tree.
+
+### Nesting Depth Limit
+
+When a JSON object or array has an element that is itself another JSON object or array, that inner object or array is 
+said to “nest” within the outer object or array. To avoid stack overflow, it's good to have a limit on the nesting depth.
+The default path limit is 128 levels. Any attempt to create a document that contains a nesting depth greater than 128 
+will be rejected with an error message. The value can be overridden by module config json.max-path-limit.
+
+### Module API
+
+ValkeyJSON integrates with Valkey core by implementing Valkey module API.
+
+![](./images/json_overview.png)
```

Comment:
I am uncomfortable with this diagram.  The replica seems to me to be superfluous to the information being communicated.  The `JSON Module` looks like it is part of Valkey when modules created via the Module API are optional and fully external to Valkey.

Reply:
Removed the diagram.